app/services/policies/policies_list_service.rb

```diff
@@ -61,4 +66,20 @@ def to_h
       @sets.transform_values(&:to_a).as_json
     end
   end
+
+  HTTP_ERRORS = [HTTPClient::BadResponseError, HTTPClient::TimeoutError, HTTPClient::ConnectTimeoutError, HTTPClient::SendTimeoutError, HTTPClient::ReceiveTimeoutError, SocketError].freeze
+  private_constant :HTTP_ERRORS
+
+  def self.fetch_policies_from_apicast
+    apicast_registry_url = ThreeScale.config.sandbox_proxy.apicast_registry_url
+    begin
+      response = ::JSONClient.get(apicast_registry_url)
+      return response.body['policies'] if response.ok?
+      raise PoliciesListServiceError, "APIcast could not be found at url: '#{apicast_registry_url}'. Error: #{response.content}"
```

Comment:
mmm... It feels strange having this line almost exactly the same twice 🤔 maybe there is a way that we could have it only once. And the text should probably be in a locale

Reply:
It's a bit weird yes. The important difference is one throws the error from JSONClient, the other from the http response. Firstly it was rescuing `::JSONClient.get()` instead of the whole method but this way it looks a bit better. Let's think about it.

Reply:
Put it in a method

```ruby
def self.raise_policies_list_error(message)
  raise PoliciesListServiceError, I18n.t('errors.messages.apicast_not_found', url: apicast_registry_url, error: message)
end
```